pages/tokens.vue

```diff
@@ -308,4 +371,89 @@
 			sendErrorToast(e);
 		}
 	};
+
+	// APPLICATIONS
+
+	const applications = ref<Application[]>([]);
+	const applicationsCount = ref(0);
+	const loadingApplications = ref(false);
+	const firstApp = ref(0);
+
+	const onAppsPage = (event: PageState) => {
+		firstApp.value = event.first;
+		loadApplications();
+	};
+
+	const loadApplications = async () => {
+		if (!auth.isAdmin) {
+			return;
+		}
+
+		loadingApplications.value = true;
+
+		try {
+			const [ gpApps, [{ count }] ] = await Promise.all([
+				$directus.request(readItems('gp_apps', {
+					offset: firstApp.value,
+					limit: itemsPerPage,
+					sort: '-date_created',
+				})),
+				$directus.request<[{count: number}]>(aggregate('gp_apps', {
+					aggregate: { count: '*' },
+				})),
+			]);
+
+			const appTokens = await $directus.request(readItems('gp_tokens', {
+				filter: { app_id: { _in: gpApps.map(app => app.id) } },
+				sort: '-date_last_used',
+			}));
+			const apps = gpApps.map((app) => {
+				const token = appTokens.find(token => token.app_id === app.id);
+				return {
+					...app,
+					date_last_used: token ? token.date_last_used : null,
+					owner_name: app.owner_name || 'Globalping',
+				};
+			});
+
+			applications.value = apps;
+			applicationsCount.value = count;
+		} catch (e) {
+			sendErrorToast(e);
+		}
+
+		loadingApplications.value = false;
+	};
+
+	onMounted(async () => {
+		await loadApplications();
+	});
+
+	// REVOKE APP ACCESS
+
+	const revokeDialog = ref(false);
+	const appToRevoke = ref<Application | null>(null);
+
+	const openRevokeDialog = (id: string) => {
+		const token = applications.value.find(application => application.id === id);
+		appToRevoke.value = token!;
+		revokeDialog.value = true;
+	};
+
+	const revokeApp = async () => {
```

Comment:
I can see the requests being sent here, no errors, success toast message, but the apps don't get deleted (even after page reload).